src/views/tutorial-view/server.ts

```diff
@@ -137,44 +137,52 @@ const moizeOpts: Options = {
 // limit the expensive call for tutorials that have the same product
 const cachedGetAllCollections = moize(getAllCollections, moizeOpts)
 
-export async function getTutorialPagePaths(
-	product: ProductOption
-): Promise<TutorialPagePaths[]> {
-	const allCollections = await cachedGetAllCollections({
-		product: { slug: product },
-	})
-	// Only build collections where this product is the main 'theme'
-	// @TODO once we implement the `theme` query option, remove the theme filtering
-	// https://app.asana.com/0/1201903760348480/1201932088801131/f
-	const filteredCollections = allCollections.filter((c) => c.theme === product)
-	// go through all collections, get the collection slug
-	const paths = filteredCollections.flatMap((collection) => {
-		const collectionSlug = splitProductFromFilename(collection.slug)
-		// go through the tutorials within this collection, create a path for each
-		return collection.tutorials.map((tutorial) => {
-			const tutorialSlug = splitProductFromFilename(tutorial.slug)
+export async function getTutorialPagePaths(): Promise<TutorialPagePaths[]> {
+	const allCollections = await cachedGetAllCollections()
+	let paths = []
+	// eventually this will be the whole product list? we need to build from
+	// a product-first standpoint here to ensure the routing is correct as each
+	// page needs to be build from its product-parent
+	__config.dev_dot.beta_product_slugs.map((product) => {
+		// Only build collections where this product is the main 'theme'
+		// @TODO once we implement the `theme` query option, remove the theme filtering
+		// https://app.asana.com/0/1201903760348480/1201932088801131/f
+		// @TODO we may want to adjust this to build based on the collection slug instead of the theme?
```

Comment:
Need to think this through

Reply:
This may bite us when we implement cloud, where the cloud theme isn't used.

Reply:
Maybe I should use the directory structure instead. And build paths with the collection slug that have the same product.